internal/services/environment/api_environment.go

```diff
@@ -290,17 +290,25 @@ func (client *Client) DeleteEnvironment(ctx context.Context, environmentId strin
 		Message: "Deleted using Power Platform Terraform Provider",
 	}
 
-	response, err := client.Api.Execute(ctx, nil, "DELETE", apiUrl.String(), nil, environmentDelete, []int{http.StatusAccepted, http.StatusConflict}, nil)
-	if err != nil {
-		if response.HttpResponse.StatusCode == http.StatusConflict {
-			return customerrors.WrapIntoProviderError(err, customerrors.ERROR_ENVIRONMENT_OPERATION_CONFLICT, "operation lifecycle already in progress")
-		}
-		var httpError *customerrors.UnexpectedHttpStatusCodeError
-		if errors.As(err, &httpError) {
-			return fmt.Errorf("Unexpected HTTP Status %s; Body: %s", httpError.StatusText, httpError.Body)
+	response, err := client.Api.Execute(ctx, nil, "DELETE", apiUrl.String(), nil, environmentDelete, []int{http.StatusNoContent, http.StatusAccepted, http.StatusConflict}, nil)
+
+	if response.HttpResponse.StatusCode == http.StatusNoContent {
+		return customerrors.WrapIntoProviderError(err, customerrors.ERROR_OBJECT_NOT_FOUND, "environment not found")
```

Comment:
delete should not return an error for 404.  mapping 404 to success should be done in this function and not passed back to resource.

Reply:
removed, thanks for pointing out